### DISCUSSION TITLE
Inline vs. Chunk Execution of `R` Code: How to check from within a function?

### DISCUSSION THREAD

**Discussion by r-cheologist** [quarto-dev/quarto-cli#6542] [Q&A]
### Description

Output of an `R` function that is to be used in both inline and chunk code execution needs (in the context of `output: asis` as a chunk execution option) to use `cat()` only in the latter case.

It would thus be convenient to be able to achieve what the following pdeudo-code demonstrates:

```r
myfun <- function(x)
{
  if (IS_EXECUTED_INLINE) return(x) else cat(x)
}
```

Is there functionality available that provides access to the `IS_EXECUTED_INLINE` bit of information?

**Comment by mcanouil** [comment]
This is not up to Quarto (or rmarkdown for that matter).
You have to handle this yourself I am afraid.
You could register a new print method, knitr hooks, etc. to do what you want, see the "RMarkdown definitive guide" book for these.

- <https://bookdown.org/yihui/rmarkdown/>.

**Comment by r-cheologist** [comment]
Thank you, @mcanouil  for sending me down the right path.

Exploring the output of `knitr::knit_hooks$get("inline")` &  `knitr::knit_hooks$get("chunk")` led me to adding the following to my function, which should bring the output of any chunk in line with that of inline code:

```r
knitr::knit_hooks$set(
    chunk  = function(x, options) {
      x %<>% stringi::stri_replace_first_regex(
        "^\\[\\d+\\]\\s*\"(.*)\"\\s*$", "$1")
      paste(as.character(x), collapse = ", ")
    })
on.exit(knitr::knit_hooks$restore())
```

This appears to do what I want.

**Comment by gadenbuie** [comment]
@r-cheologist Have you tried using `knitr::asis_output(x)`? With knitr, this is typically the best way to consistently return text that should be added literally into the document without having to resort to using `cat()`.

**Reply by r-cheologist** [reply]
Interesting hint and I played with this for a long while, but see [here](https://github.com/quarto-dev/quarto-cli/discussions/6874#discussion-5643032): `quarto` is generating interfering enclosing containers when this approach is used.

**Comment by cderv** [comment]
Adding that `knit_print` method is usually a good way to have different behavior for inline vs in-chunk as there is `inline` argument set to TRUE when inline. 

See more at https://cran.r-project.org/web/packages/knitr/vignettes/knit_print.html
